ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -119,6 +123,10 @@ public void run(MLAgent mlAgent, Map<String, String> params, ActionListener<Obje
                 Iterator<Interaction> iterator = r.iterator();
                 while (iterator.hasNext()) {
                     Interaction next = iterator.next();
+                    if (next.getId().equals(regenerateInteractionId)) {
+                        // there may have other new interactions happened after this interaction
```

Comment:
Could you update the comment to be more clear?

Reply:
sure. comments added.